sdk/identity/Azure.Identity/src/ScopeGroupHandler.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Azure.Core.Pipeline;
+
+namespace Azure.Identity
+{
+    internal sealed class ScopeGroupHandler : IScopeHandler
+    {
+        private static readonly AsyncLocal<IScopeHandler> _currentAsyncLocal = new AsyncLocal<IScopeHandler>();
+        private readonly string _groupName;
+        private Dictionary<string, ChildScopeInfo> _childScopes;
+
+        public static IScopeHandler Current => _currentAsyncLocal.Value;
```

Comment:
I had to return it back to the `AsyncLocal<IScopeHandler>` implementation because in `ChainedTokenCredential` there is no way to control if underlying sources are used in parallel. `ScopeGroupHandler` should be unique per call of `GetToken`, at the same time we can't add parameters to the method signature.